packages/arb-bridge-eth/contracts/rollup/Rollup.sol

```diff
@@ -158,6 +169,27 @@ contract Rollup is RollupBase {
         require(success, "FAIL_INIT_FACET");
 
         emit RollupCreated(_machineHash);
+        require(isInit(), "INITIALIZE_NOT_INIT");
+    }
+
+    function postUpgradeInit(address newAdminFacet) external {
```

Comment:
Should we put `postUpgradeInit` into the admin facet? Also we'll definitely need to upgradeAndCall when we do this upgrade since the init hook takes input

Reply:
This is currently on the Rollup dispatch which I believe works well.
I agree this should be triggered in the `upgradeAndCall`, but that is done by the ProxyAdmin, which is different to the Rollup Owner that has access to the admin facet.